### ISSUE TITLE
How to have custom action [[ ]] / wiki style links

### ISSUE THREAD

**Issue by damionx7** [noties/Markwon#323]
* **Markwon version**:  4.6.0

Hey noties . Thanks for this library. Its really not an issue but there is one thing I want to ask. 

1. How to handle wiki style links eg. [[link]] .

**Comment by damionx7** [comment]
Got it. I solved this by using following code

```
Pattern source = Pattern.compile("\\[\\[(.*)]]", Pattern.MULTILINE);
markwon.setMarkdown(mTextViewNoteContent, noteModel.getContent());
Linkify.addLinks(mTextViewNoteContent, source,"");
mTextViewNoteContent.setMovementMethod(BetterLinkMovementMethod.newInstance().setOnLinkClickListener(new BetterLinkMovementMethod.OnLinkClickListener() {
      @Override
      public boolean onClick(TextView textView, String url) {
        if(url.startsWith("[[")){
          for(Listener listener : getListeners()){
            listener.onInlineUrlClick(url);
          }
          return true;
        }
        return false;
      }
    }));
```

**Comment by noties** [comment]
Hello @damionx7 ,

it's an interesting way of achieving that. The only thing that I see is that your regex is not lazy - multiple matches on the same line would be matched as one:

```markdown
[[wiki]] [[kiwi]]
```

You should change your pattern to `\\[\\[.*?]]` instead

**Comment by damionx7** [comment]
Thanks @noties . Yes, I figured that out later on.

**Comment by damionx7** [comment]
Hey there @noties . When i use the method of Linkify.addLinks, the `builder.appendFactory(Link.class, ((configuration, props) -> new ForegroundColorSpan(Color.MAGENTA)));` color is removed.

Is there any way to add custom / regex linkify links via markwon ?

**Comment by noties** [comment]
Hello @damionx7 ,

links from rendered markdown are removed by the `Linkify.addLinks` method. Inside it removes all instances of `URLSpan` from text before processing. This includes `LinkSpan` that Markwon uses by default for links. This is really strange and can be surprising from `Linkify`.

You can take a look at how `LinkifyPlugin` deals with such behavior - it calls `Linkify.addLinks` with a different text buffer and then moves added links to the main buffer.

You can also ignore the `Linkify` and add your links manually - as you have a non-standard format and you have a working regex for it. So you actually do not use that much of what `Linkify` offers. Instead you can create a `Matcher` and add all the links yourself by choosing whatever span you need. 

I think it would be a good thing also to escape the `[[` sequence by turning it into `\[\[` before passing text to `Markwon` for parsing. For example in a `MarkwonPlugin` by overriding `String processMarkdown(@NonNull String markdown)` method.

You could also use that method to change all `[[link]]` occurrences into a regular markdown link form: `[link](#link)` and then handle the clicks in a custom `LinkResolver`.

In short, there are multiple possibilities to render additional links. Just don't use `Linkify` or use with caution and mind that it removes all the links first and only after that adds links configured
